homeassistant/components/climate/opentherm_gw.py

```diff
@@ -0,0 +1,187 @@
+                                              STATE_COOL,
+                                              SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import (ATTR_TEMPERATURE, CONF_DEVICE, CONF_NAME,
+                                 PRECISION_HALVES, PRECISION_TENTHS,
+                                 TEMP_CELSIUS, PRECISION_WHOLE)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pyotgw==0.1b0']
+
+CONF_FLOOR_TEMP = "floor_temperature"
+CONF_PRECISION = 'precision'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Optional(CONF_NAME, default="OpenTherm Gateway"): cv.string,
+    vol.Optional(CONF_PRECISION): vol.In([PRECISION_TENTHS, PRECISION_HALVES,
+                                          PRECISION_WHOLE]),
+    vol.Optional(CONF_FLOOR_TEMP, default=False): cv.boolean,
+})
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE)
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the opentherm_gw device."""
+    gateway = OpenThermGateway(config)
+    async_add_entities([gateway])
+
+
+class OpenThermGateway(ClimateDevice):
+    """Representation of a climate device."""
+
+    def __init__(self, config):
+        """Initialize the sensor."""
+        import pyotgw
+        self.pyotgw = pyotgw
+        self.gateway = self.pyotgw.pyotgw()
+        self._device = config.get(CONF_DEVICE)
+        self.friendly_name = config.get(CONF_NAME)
+        self.floor_temp = config.get(CONF_FLOOR_TEMP)
+        self.temp_precision = config.get(CONF_PRECISION)
+        self._current_operation = STATE_IDLE
+        self._current_temperature = 0.0
+        self._target_temperature = 0.0
+        self._away_mode_a = None
+        self._away_mode_b = None
+        self._away_state_a = False
+        self._away_state_b = False
+
+    async def async_added_to_hass(self):
+        """Connect to the OpenTherm Gateway device."""
+        await self.gateway.connect(self.hass.loop, self._device)
+        self.gateway.subscribe(self.receive_report)
+        _LOGGER.debug("Connected to %s on %s", self.friendly_name,
+                      self._device)
+
+    async def receive_report(self, status):
+        """Receive and handle a new report from the Gateway."""
+        _LOGGER.debug("Received report: %s", status)
+        ch_active = status.get(self.pyotgw.DATA_SLAVE_CH_ACTIVE)
+        cooling_active = status.get(self.pyotgw.DATA_SLAVE_COOLING_ACTIVE)
+        if ch_active:
+            self._current_operation = STATE_HEAT
+        elif cooling_active:
+            self._current_operation = STATE_COOL
+        else:
+            self._current_operation = STATE_IDLE
+        self._current_temperature = status.get(self.pyotgw.DATA_ROOM_TEMP)
+        if status.get(self.pyotgw.DATA_ROOM_SETPOINT_OVRD) is None:
```

Comment:
```py
temp = status.get(self.pyotgw.DATA_ROOM_SETPOINT_OVRD)
if temp is None:
    temp = status.get(self.pyotgw.DATA_ROOM_SETPOINT)
self._target_temperature = temp
```

Reply:
This optimization can be applied in other places as well. Will do in the next commit!

Reply:
Addressed in105f8eb668b11445d0de0862f32aebb783b9e3c1